### ISSUE TITLE
RStudio: Make sure Environment pane is set to 'Manual Refresh Only' when needed

### ISSUE THREAD

**Issue by HenrikBengtsson** [HenrikBengtsson/trackers#4]
# Idea

Add a tracker that detects when an `iter` object is being added to the global environment:

```r
> library(iterators)
> x <- iter(1:3)
> class(x)
[1] "containeriter" "iter"
```

_and_ when we're running RStudio and the RStudio Environment pane is not set to "Manual Refresh Only", because then RStudio calls `str(x)` in the background, which causes the iterator to be forwarded as if:

```r
> x$state$i
[1] 0

> str(x)
List of 4
 $ state    :<environment: 0x557bea11d740> 
 $ length   : int 3
 $ checkFunc:function (...)  
 $ recycle  :Error in oTypes[[i]] : subscript out of bounds

> x$state$i
[1] 4
```

The tracker could warn, or even produce an error, saying that `iter` objects are not compatible with RStudio unless "Manual Refresh Only" is checked in the Environment pane.

# See also

* https://github.com/RevolutionAnalytics/iterators/issues/3
* https://github.com/RevolutionAnalytics/iterators/issues/4

**Comment by binary10** [comment]
Hi Henrik, thanks for your attention to this issue. Would it be possible to correct this issue by changing the behavior of the `str` built-in function? This is the recommendation in the SO post I shared in my issue. Whether one is working with an iterator in RStudio or in another env, as a user I would not expect `str` to advance the iterator. The purpose of `str` in the docs is to "Compactly Display the Structure of an Arbitrary R Object", and I would expect this to be possible without side effects.

**Comment by HenrikBengtsson** [comment]
Yes, that sounds like a workaround. I can also imagine that Posit/RStudio could keep a list of object classes that should be excluded from auto-str() calling.

PS. Note that I am not affiliated with this package, so I cannot make any updates.
